pkg/vault/test/foundry/fork/YieldBearingPoolSwapBase.t.sol

```diff
@@ -0,0 +1,1097 @@
+        vars.bufferBeforeSwapYbTokenOut = wrappedBalance;
+
+        uint256[] memory balancesRaw;
+        (, , balancesRaw, ) = vault.getPoolTokenInfo(address(yieldBearingPool));
+        if (vars.tokenIn == _token1Fork) {
+            vars.indexYbTokenIn = _ybToken1Idx;
+            vars.indexYbTokenOut = _ybToken2Idx;
+        } else {
+            vars.indexYbTokenIn = _ybToken2Idx;
+            vars.indexYbTokenOut = _ybToken1Idx;
+        }
+        vars.poolBeforeSwapYbTokenIn = balancesRaw[vars.indexYbTokenIn];
+        vars.poolBeforeSwapYbTokenOut = balancesRaw[vars.indexYbTokenOut];
+    }
+
+    function _buildExactInPaths(
+        IERC20 tokenIn,
+        uint256 exactAmountIn,
+        uint256 minAmountOut
+    ) private view returns (IBatchRouter.SwapPathExactAmountIn[] memory paths) {
+        paths = new IBatchRouter.SwapPathExactAmountIn[](1);
+        paths[0] = IBatchRouter.SwapPathExactAmountIn({
+            tokenIn: tokenIn,
+            steps: _getSwapSteps(tokenIn),
+            exactAmountIn: exactAmountIn,
+            minAmountOut: minAmountOut
+        });
+    }
+
+    function _buildExactOutPaths(
+        IERC20 tokenIn,
+        uint256 maxAmountIn,
+        uint256 exactAmountOut
+    ) private view returns (IBatchRouter.SwapPathExactAmountOut[] memory paths) {
+        paths = new IBatchRouter.SwapPathExactAmountOut[](1);
+        paths[0] = IBatchRouter.SwapPathExactAmountOut({
+            tokenIn: tokenIn,
+            steps: _getSwapSteps(tokenIn),
+            maxAmountIn: maxAmountIn,
+            exactAmountOut: exactAmountOut
+        });
+    }
+
+    function _getSwapSteps(IERC20 tokenIn) private view returns (IBatchRouter.SwapPathStep[] memory steps) {
+        steps = new IBatchRouter.SwapPathStep[](3);
+
+        if (tokenIn == _token2Fork) {
+            steps[0] = IBatchRouter.SwapPathStep({
+                pool: address(ybToken2),
+                tokenOut: IERC20(address(ybToken2)),
+                isBuffer: true
+            });
+            steps[1] = IBatchRouter.SwapPathStep({
+                pool: address(yieldBearingPool),
+                tokenOut: IERC20(address(ybToken1)),
+                isBuffer: false
+            });
+            steps[2] = IBatchRouter.SwapPathStep({ pool: address(ybToken1), tokenOut: _token1Fork, isBuffer: true });
+        } else {
+            steps[0] = IBatchRouter.SwapPathStep({
+                pool: address(ybToken1),
+                tokenOut: IERC20(address(ybToken1)),
+                isBuffer: true
+            });
+            steps[1] = IBatchRouter.SwapPathStep({
+                pool: address(yieldBearingPool),
+                tokenOut: IERC20(address(ybToken2)),
+                isBuffer: false
+            });
+            steps[2] = IBatchRouter.SwapPathStep({ pool: address(ybToken2), tokenOut: _token2Fork, isBuffer: true });
+        }
+    }
+
+    function _unbalanceBuffers(uint256 unbalancedToken1, uint256 unbalancedToken2) private {
```

Comment:
Can we add a very small `@dev` block explaining that this can unbalance the buffer to either side of the equilibrium point?